src/main/fc/fc_msp.c

```diff
@@ -1892,6 +1893,150 @@ static mspResult_e mspFcProcessInCommand(uint16_t cmdMSP, sbuf_t *src)
     return MSP_RESULT_ACK;
 }
 
+static const setting_t *mspReadSettingName(sbuf_t *src)
+{
+    char name[SETTING_MAX_NAME_LENGTH];
+    uint8_t c;
+    size_t s = 0;
+    while (1) {
```

Comment:
As MSP will give you the payload size, why not just use that instead of requiring a NUL terminating byte (the consumer has to supply the payload size regardless).

Reply:
Huum, I'd not considered the set use case, where the sum of name and value is somewhat less determinate.

Reply:
@stronagg that’s right, you’re in a catch 22 without an explicit terminator or name length. You need to know the setting name in order to find how big it’s data is. The getter could do without the zero terminator, but I thought using the same encoding for getter and setter would be more intuitive. I’d expect these new commands to be used mostly for configuration and not for live updates, so I thought the extra byte would not make a difference.